lib/ansible/modules/network/panos/panos_pg.py

```diff
@@ -0,0 +1,233 @@
+  panos_pg:
+    ip_address: "192.168.1.1"
+    password: "admin"
+    username: "admin"
+    pg_name: "pg-default"
+    virus: "default"
+    spyware: "default"
+    vulnerability: "default"
+'''
+
+RETURN = '''
+status:
+    description: success status
+    returned: success
+    type: string
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+try:
+    import pan.xapi
+    from pan.xapi import PanXapiError
+    HAS_LIB = True
+except ImportError:
+    HAS_LIB = False
+
+_PG_XPATH = "/config/devices/entry[@name='localhost.localdomain']" +\
+            "/vsys/entry[@name='vsys1']" +\
+            "/profile-group/entry[@name='%s']"
+
+
+def pg_exists(xapi, pg_name):
+    xapi.get(_PG_XPATH % pg_name)
+    e = xapi.element_root.find('.//entry')
+    if e is None:
+        return False
+    return True
+
+
+def add_pg(xapi, pg_name, data_filtering, file_blocking, spyware,
+           url_filtering, virus, vulnerability, wildfire):
+    if pg_exists(xapi, pg_name):
+        return False
+
+    exml = []
+
+    if data_filtering is not None:
+        exml.append('<data-filtering><member>%s</member></data-filtering>' %
+                    data_filtering)
+    if file_blocking is not None:
+        exml.append('<file-blocking><member>%s</member></file-blocking>' %
+                    file_blocking)
+    if spyware is not None:
+        exml.append('<spyware><member>%s</member></spyware>' %
+                    spyware)
+    if url_filtering is not None:
+        exml.append('<url-filtering><member>%s</member></url-filtering>' %
+                    url_filtering)
+    if virus is not None:
+        exml.append('<virus><member>%s</member></virus>' %
+                    virus)
+    if vulnerability is not None:
+        exml.append('<vulnerability><member>%s</member></vulnerability>' %
+                    vulnerability)
+    if wildfire is not None:
+        exml.append('<wildfire-analysis><member>%s</member></wildfire-analysis>' %
+                    wildfire)
+
+    exml = ''.join(exml)
+    xapi.set(xpath=_PG_XPATH % pg_name, element=exml)
+
+    return True
+
+
+def main():
+    argument_spec = dict(
+        ip_address=dict(required=True),
+        password=dict(required=True, no_log=True),
+        username=dict(default='admin'),
+        pg_name=dict(required=True),
+        data_filtering=dict(),
+        file_blocking=dict(),
+        spyware=dict(),
+        url_filtering=dict(),
+        virus=dict(),
+        vulnerability=dict(),
+        wildfire=dict(),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```

Comment:
not needed.  setting `required=True` in the argument_spec will automatically generate an failed message if the argument is not specified.